### PR TITLE
Support RocksDB 7.x BackupEngineOptions

### PR DESCRIPTION
**Pull request by exabytes18**
The main objective of this PR is to support `rocksdb_backup_engine_options_t` (introduced in https://github.com/facebook/rocksdb/commit/78aee6fedc62ca6048e500969912d1e1f0553490) so that we can call the associated `rocksdb_backup_engine_options_set_max_background_operations` function and achieve higher parallelism for backup and restore operations (notably, the current parallelism is 1, meaning that rocksdb will only copy/checksum/hardlink/whatever 1 file at a time, which is insufficient for rocksdb to full utilize hardware on even modest machines).

One side-effect of this PR is that the signature for BackupEngine::open changes from `open<P: AsRef<Path>>(opts: &BackupEngineOptions, path: P)` to `open(opts: &BackupEngineOptions, env: &Env)`. This may cause some consumer code to fail to compile, but it aligns better with the RocksDB-proper `BackupEngine::Open()` signature, so I believe it's a worthwhile change.

Finally, I'm not well-versed in ffi or the related patterns (like `_outlive`), so please carefully review those bits in the PR.

Thanks!
Matt

### CODE REVIEW

**Comment by exabytes18** [comment]
@aleksuss, @stanislav-tkach, could you help to review this PR?

**Comment by exabytes18** [comment]
@aleksuss, I've fixed a clippy warning. Could you approve the workflow run again?

**Comment by exabytes18** [comment]
@aleksuss, could you take a look at the PR when you next have a chance?

**Review by stanislav-tkach — APPROVED** [review]

**Comment by exabytes18** [comment]
@aleksuss, @stanislav-tkach, could you help to merge this?

File: src/backup.rs [hunk 1]
```diff
@@ -217,27 +224,40 @@ impl BackupEngine {
 }
 
 impl BackupEngineOptions {
-    //
-}
+    pub fn new<P: AsRef<Path>>(backup_dir: P) -> Result<Self, Error> {
```
Comment (aleksuss) [inline src/backup.rs:227]: Please, add some description to the public methods.

Comment (exabytes18) [reply src/backup.rs:229]: Done. Please have another look.

**Review by aleksuss — APPROVED** [review]

**Pull request merged**